### PR TITLE
Add factory for `PageChooserBlock` and `DocumentChooserBlock`

### PR DESCRIPTION
**Pull request by Tijani-Dia**
This PR adds a `PageChooserBlockfactory` and a `DocumentChooserBlockFactory` similar to the already existing `ImageChooserBlockFactory`.

### CODE REVIEW

**Review by bcdickinson — CHANGES_REQUESTED** [review]
Thanks for this, it's looking good! I do have some questions about the utility of the `PageChooserBlock`, though, as it's not likely it will be useful in real projects to create bare `Page` instances - you almost always want a subclass. 

My assumption is that for this to be useful you'd have to subclass `PageChooserBlockFactory` and override the `page` declaration to provide your own page subclass (via a `SubFactory` or otherwise). Can we add some tests (and maybe even update the example in the README?!) to show that this usage is possible and works correctly?

**Comment by Tijani-Dia** [comment]
Thanks! I've updated the tests and docs to show the use case you mentioned.

File: tests/test_blocks.py [hunk 1]
```diff
@@ -153,9 +170,25 @@ def test_custom_page_streamfield_default_blocks():
...
 
     assert value == image
+
+
+@pytest.mark.django_db
+def test_document_chooser_block():
+    value = wagtail_factories.DocumentChooserBlockFactory()
+    document = Document.objects.last()
+
+    assert value == document
```
Comment (jams2) [inline tests/test_blocks.py:194]: One thing that would be good to address in these tests is whether or not the chooser block factories respect the [build strategy](https://factoryboy.readthedocs.io/en/stable/introduction.html#strategies) provided by the user. Can we add explicit tests for `.build` and `.create`, and test that the subfactory object is only persisted to the DB when the strategy is `create`?

Comment (Tijani-Dia) [reply tests/test_blocks.py]: Thanks @jams2. I've added tests for the strategies.

**Review by bcdickinson — APPROVED** [review]

File: README.rst [hunk 2]
```diff
@@ -63,10 +63,19 @@ Documentation is still in progress, but see the `tests`_ for more examples.
...
+
+    class MyNewsPageChooserBlockFactory(wagtail_factories.PageChooserBlockFactory):
+        page = factory.SubFactory(MyNewsPageFactory)
 
+
+    class MyTestPageFactory(wagtail_factories.PageFactory):
         body = wagtail_factories.StreamFieldFactory({
-            'carousel': MyCarouselFactory
+            'carousel': MyCarouselFactory,
+            'news_page': MyNewsPageChooserBlockFactory,
```
Comment (jams2) [inline README.rst:78]: @Tijani-Dia I missed this before sorry:

```suggestion
            'carousel': factory.SubFactory(MyCarouselFactory),
            'news_page': factory.SubFactory(MyNewsPageChooserBlockFactory),
```

Comment (Tijani-Dia) [reply README.rst]: Great, you caught it.

**Pull request merged**